include/seqan3/io/alignment_file/record.hpp

```diff
@@ -0,0 +1,415 @@
+    sam_record(sam_record const &) = default; //!< Defaulted.
+    sam_record & operator=(sam_record const &) = default; //!< Defaulted.
+    sam_record(sam_record &&) = default; //!< Defaulted.
+    sam_record & operator=(sam_record &&) = default; //!< Defaulted.
+    ~sam_record() = default; //!< Defaulted.
+
+    //!\brief Inherit std::tuple's and seqan3::record constructors.
+    using base_t::base_t;
+    //!\}
+
+    //!\brief The identifier, usually a string.
+    decltype(auto) id() &&
+    {
+        return get_impl(field_constant<seqan3::field::id>{}, static_cast<tuple_base_t &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::id
+    decltype(auto) id() const &&
+    {
+        return get_impl(field_constant<seqan3::field::id>{}, static_cast<tuple_base_t const &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::id
+    decltype(auto) id() &
+    {
+        return get_impl(field_constant<seqan3::field::id>{}, static_cast<tuple_base_t &>(*this));
+    }
+    //!\copydoc seqan3::sam_record::id
+    decltype(auto) id() const &
+    {
+        return get_impl(field_constant<seqan3::field::id>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief The "sequence", usually a range of nucleotides or amino acids.
+    decltype(auto) sequence() &&
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<tuple_base_t &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::sequence
+    decltype(auto) sequence() const &&
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<tuple_base_t const &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::sequence
+    decltype(auto) sequence() &
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<tuple_base_t &>(*this));
+    }
+    //!\copydoc seqan3::sam_record::sequence
+    decltype(auto) sequence() const &
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief The qualities, usually in phred-score notation.
+    decltype(auto) base_qualities() &&
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<tuple_base_t &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::base_qualities
+    decltype(auto) base_qualities() const &&
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<tuple_base_t const &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::base_qualities
+    decltype(auto) base_qualities() &
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<tuple_base_t &>(*this));
+    }
+    //!\copydoc seqan3::sam_record::base_qualities
+    decltype(auto) base_qualities() const &
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief Sequence (SEQ) relative start position (0-based), unsigned value.
```

Comment:
```suggestion
    //!\brief Sequence (SEQ) relative start position (0-based), unsigned value.
```

Mhm, should it link to the `sequence()` function instead of `SEQ`? Now we suddenly use SAM field notation where we used the SeqAn entities before - and we do not link to the SAM notation always either...

Reply:
```suggestion
    //!\brief Sequence (seqan3::sam_record::sequence) relative start position (0-based), unsigned value.
```